routemaster/mixins/log.rb

```diff
@@ -66,6 +70,22 @@ def _smart_backtrace(e)
           matches || !seen_own
         end
       end
+
+      def _log_level_constant
+        if _log_level_invalid?
+          Logger::INFO
+        else
+          Logger.const_get(_fetch_log_level)
```

Comment:
Since this is only called once — you could call this, rescue the `NameError` exception and log there. Something like this:

```ruby
def _log_level
  level_name = ENV.fetch('ROUTEMASTER_LOG_LEVEL', 'INFO')
  Logger.const_get level_name
rescue NameError
  warn "log level #{level_name} is invalid, defaulting to INFO"
end
```